SCons/Environment.py

```diff
@@ -2000,6 +2001,19 @@ def CacheDir(self, path):
 
     def Clean(self, targets, files):
         global CleanTargets
+
+        # Check for anything which evaluates to empty string, which would yield cleaning '.'
+        targets_strings = [(t, self.subst(t)) for t in flatten(targets) if is_String(t) and ('$' in t or t == '')]
```

Comment:
wouldn't it be simpler to just let `tlist` and `flist` expand and check in them?

Reply:
So I don't think using tlist or flist would allow proper checking for accidental '.' from undefined var.
Also you wouldn't be able to give as much information about what was wrong.